pkg/prometheus/operator.go

```diff
@@ -1000,24 +1002,36 @@ func (c *Operator) createOrUpdateConfigurationSecret(p *monitoringv1.Prometheus)
 	}
 
 	sClient := c.kclient.CoreV1().Secrets(p.Namespace)
-
-	listSecrets, err := sClient.List(metav1.ListOptions{})
-
+	promSecrets, err := sClient.List(metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	basicAuthSecrets, err := c.loadBasicAuthSecrets(smons, p.Spec.RemoteRead, p.Spec.RemoteWrite, listSecrets)
+	secretMap := make(map[string]*v1.SecretList)
```

Comment:
`secretMap` seems a bit generic to me. What do you think about `sMonSecretMap`?

Reply:
In addition it seesm like `secretMap` is only used inside of `loadBasicAuthSecrets`. How about retrieving the secrets in there, instead of having the logic distributed between functions?

Reply:
good call, let me do it